concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/AbstractLimiter.java

```diff
@@ -24,12 +24,36 @@
 
 import java.util.Optional;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
 import java.util.function.Supplier;
 
 public abstract class AbstractLimiter<ContextT> implements Limiter<ContextT> {
     public static final String ID_TAG = "id";
     public static final String STATUS_TAG = "status";
 
+    /**
+     * Constructs a new builder with a list of bypass resolvers.
+     * If the predicate condition in any of the resolver is satisfied,
+     * the call is bypassed without increasing the limiter inflight count
+     * and affecting the algorithm.
+     */
+    public abstract static class BypassLimiterBuilder<BuilderT extends BypassLimiterBuilder<BuilderT, ContextT>, ContextT> extends Builder<BuilderT> {
+
+        private Predicate<ContextT> bypassResolver = (context) -> false;
+
+        /**
+         * Add a chainable bypass resolver predicate from context. Multiple resolvers may be added and if any of the
+         * predicate condition returns true the call is bypassed without increasing the limiter inflight count and
+         * affecting the algorithm. Will not bypass any calls by default if no resolvers are added.
+         * @param shouldBypass Predicate condition to bypass limit
+         * @return Chainable builder
+         */
+        public BuilderT bypassLimitResolver(Predicate<ContextT> shouldBypass) {
+            this.bypassResolver = bypassResolver.or(shouldBypass);
```

Comment:
```suggestion
            if(this.bypassResolver == ALWAYS_FALSE) {
                this.bypassResolver = shouldBypass;
            } else {
                this.bypassResolver = bypassResolver.or(shouldBypass);
            }
```

Since this is on the hot path for calls, let's optimize away the default case instead of hoping that the JVM is able to do it later.

Reply:
This is the builder method. The else condition will be skipped everytime because `this.bypassResolver` is aways equal to ALWAYS_FALSE in your suggestion.

Reply:
Read your code wrong. Applying suggestion.